app/src/main/java/com/example/iphonesearchapi/MainActivity.kt

```diff
@@ -2,10 +2,64 @@ package com.example.iphonesearchapi
 
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.view.View
+import androidx.lifecycle.Observer
+import androidx.activity.viewModels
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.iphonesearchapi.adapter.ItunesAdapter
+import com.example.iphonesearchapi.databinding.ActivityMainBinding
+import com.example.iphonesearchapi.model.ResultOf
+import com.example.iphonesearchapi.viewmodel.ItunesViewModel
+
+private lateinit var binding: ActivityMainBinding
+const val BaseUrl = "https://itunes.apple.com/"
 
 class MainActivity : AppCompatActivity() {
+
+    private val viewModel: ItunesViewModel by viewModels()
+
+        fun showLoadingIndicator(showLoading: Boolean){
+           binding.progressBar.visibility = if (showLoading) View.VISIBLE else View.INVISIBLE
+        }
+
+
+
+    private val itunesAdapter: ItunesAdapter by lazy {
+        ItunesAdapter( mutableListOf())
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        val linearLayoutManager = LinearLayoutManager(this)
+        setContentView(binding.root)
+
+        binding.recyclerView.layoutManager = linearLayoutManager
+        binding.recyclerView.adapter = itunesAdapter
+        viewModel.itunes.observe(this,  { Resourse ->
+            when (Resourse) {
+                is ResultOf.Success -> {
```

Comment:
R in resource should be small. It is just a `val`

```suggestion
        viewModel.itunes.observe(this,  { resourse ->
            when (resourse) {
                is ResultOf.Success -> {
```